src/components/SignupLoanComponent.js

```diff
@@ -1,15 +1,97 @@
+
 import React, { Component } from "react";
+import { Formik, Field, Form, ErrorMessage, useFormik } from 'formik';
+import * as Yup from 'yup';
+import AuthService from "../services/authService";
+
+const salaryRegex = /^[0-9]*$/
+const amountRegex = /^[0-9]*$/
+const termRegex = /^[0-9]*$/
+
+
+
+const validationSchema = Yup.object().shape({
+
+    salary: Yup.string()
+    .matches(salaryRegex, 'Only numbers are allowed')
+    .required('Salary is required'),
+
+    amount: Yup.string()
+    .matches(amountRegex, 'Only numbers are allowed')
+    .required('Amount is required'),
+
+    term: Yup.string()
+        .matches(termRegex, 'Only numbers are allowed')
+        .required('term is required'),
+   
+})
```

Comment:
Pretty decent way to do it, but an explicit check for "0" and a limit as to the value of the expression (i.e. how large it can be) would probably be wise as we don't want million-year loans.

Reply:
fixed